src/pages/home/uploadSong/UploadSong.tsx

```diff
@@ -24,8 +24,14 @@ import MintSong from "./MintSong";
 const UploadSong: FunctionComponent = () => {
   const dispatch = useDispatch();
 
-  const { genres } = useSelector(selectContent);
-  const isLoading = useSelector(selectSongIsLoading);
+  const { data = [] } = useGetSongGenresQuery();
+  const genres: string[] = data.filter((g) => {
+    // Filter out empty genre
+    if (g.length > 0) {
+      return g;
+    }
+  });
```

Comment:
You can simplify this to:
`const genres = useGetSongGenresQuery().data.filter(genre => genre.length > 0);`
if you are not a fan of the one liners you can do this:

```
const { data = [] } = useGetSongGenresQuery();
const genres = data.filter(genre => genre.length > 0);

```
  
 A side note to the `g` param, as you iterate over things where you may not care too much about the single item, it makes it easier to read the code if you name it what it is that way you don't need to add the comment and it is self documented.

Reply:
I switched to your two line suggestion, thanks!